### PR TITLE
Add target counter in status bar

### PR DESCRIPTION
**Pull request by qjqqyy**
In order to determine how many entries were returned by a `find` command, or how many entries there are in the addressbook in total, the user has to scroll down.

We modify StatusBarFooter to show the total number of persons displayed in the view.

### CODE REVIEW

File: src/main/java/seedu/address/ui/MainWindow.java [hunk 1]
```diff
@@ -121,7 +121,8 @@ void fillInnerParts() {
         resultDisplay = new ResultDisplay();
         resultDisplayPlaceholder.getChildren().add(resultDisplay.getRoot());
 
-        StatusBarFooter statusBarFooter = new StatusBarFooter(logic.getAddressBookFilePath(), logic.getAddressBook());
+        StatusBarFooter statusBarFooter = new StatusBarFooter(logic.getAddressBookFilePath(), logic.getAddressBook(),
+                logic.getFilteredPersonList());
```
Comment (thomastanck) [inline src/main/java/seedu/address/ui/MainWindow.java:125]: Huh, I'm surprised this passes checkstyle. Oh well :P

File: src/main/java/seedu/address/ui/StatusBarFooter.java [hunk 2]
```diff
@@ -34,15 +38,27 @@
...
+    }
+
+    /**
+     * Updates the target count whenever the displayed person list changes.
+     */
+    private void updateTargetCount(ListChangeListener.Change<? extends Person> change) {
+        targetCountStatus.setText(String.format(TARGET_COUNT_STATUS, change.getList().size()));
     }
 
+
```
Comment (thomastanck) [inline src/main/java/seedu/address/ui/StatusBarFooter.java]: Extra newline

File: src/main/java/seedu/address/ui/StatusBarFooter.java [hunk 3]
```diff
@@ -17,6 +20,7 @@
 
     public static final String SYNC_STATUS_INITIAL = "Not updated yet in this session";
     public static final String SYNC_STATUS_UPDATED = "Last Updated: %s";
+    public static final String TARGET_COUNT_STATUS = "%d targets";
```
Comment (thomastanck) [inline src/main/java/seedu/address/ui/StatusBarFooter.java]: Maybe a better string would be "Found %d persons"?

**Review by thomastanck — DISMISSED** [review]
You need tests! Add asserts to `assertStatusBarContent` in StatusBarFooterTest.java at the minimum. Also look at AddressBookSystemTests and find for StatusBar, you'll want to add the corresponding asserts.

File: src/main/java/seedu/address/ui/StatusBarFooter.java (hunk 3 above)
Comment (qjqqyy) [reply src/main/java/seedu/address/ui/StatusBarFooter.java]: Nice suggestion indeed!

File: src/main/java/seedu/address/ui/StatusBarFooter.java (hunk 2 above)
Comment (qjqqyy) [reply src/main/java/seedu/address/ui/StatusBarFooter.java]: Nice find! Fixed, thanks!

File: src/main/java/seedu/address/ui/MainWindow.java (hunk 1 above)
Comment (qjqqyy) [reply src/main/java/seedu/address/ui/MainWindow.java:125]: This line breaking is to stay consistent with a few lines before. Although a readability issue, maybe we fix in another PR?

**Review by rlrh — APPROVED** [review]

**Pull request merged**